uccb/src/bin/validator-oracle.rs

```diff
@@ -0,0 +1,273 @@
+    config_file: PathBuf,
+}
+
+impl uccb::Args for Args {
+    fn secret_key(&self) -> &SecretKey {
+        &self.secret_key
+    }
+
+    fn config_file(&self) -> &PathBuf {
+        &self.config_file
+    }
+}
+
+// Workaround for displaying the collection of validators
+// using Display instead of Debug which shows too mube info...
+struct Display<'a>(&'a std::vec::Vec<Address>);
+
+impl<'a> std::fmt::Display for Display<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        if let Ok(s) = serde_json::to_string(self.0) {
+            write!(f, "{}", &s)?
+        }
+        Ok(())
+    }
+}
+
+struct ValidatorOracle {
+    signer: PrivateKeySigner,
+    chain_clients: Vec<ChainClient>,
+    deploy_abi: JsonAbi,
+    validator_manager_abi: JsonAbi,
+}
+
+impl ValidatorOracle {
+    pub async fn new(secret_key: SecretKey, config: Config) -> Result<Self> {
+        let signer = PrivateKeySigner::from_str(secret_key.to_hex().as_str())?;
+        let mut chain_clients = vec![Self::create_zq2_chain_client(config.clone(), &signer).await?];
+
+        for chain_config in config.chain_configs {
+            chain_clients.push(
+                ChainClient::new(
+                    &chain_config,
+                    config.zq2.validator_manager_address,
+                    signer.clone(),
+                )
+                .await?,
+            );
+        }
+
+        Ok(Self {
+            signer,
+            chain_clients,
+            deploy_abi: serde_json::from_value(serde_json::to_value(
+                contracts::deposit::ABI.clone(),
+            )?)?,
+            validator_manager_abi: serde_json::from_value(
+                serde_json::from_str::<serde_json::Value>(VALIDATOR_MANAGER_ABI_JSON)?["abi"]
+                    .clone(),
+            )?,
+        })
+    }
+
+    pub async fn start(&mut self) -> Result<()> {
+        info!(
+            "Starting validator oracle with signer address {}",
+            self.signer.address()
+        );
+
+        let validators = self.get_stakers().await?;
+        info!("Current validator set is: {}", Display(&validators));
+
+        let (sender, receiver) = watch::channel(validators);
+
+        let mut handles = vec![];
+        for chain_client in &self.chain_clients {
+            let mut receiver = receiver.clone();
+            let chain_client = chain_client.clone();
+            let validator_manager_abi = self.validator_manager_abi.clone();
+            let handle = tokio::spawn(async move {
+                loop {
+                    let validators = receiver.borrow_and_update().clone();
+                    if Self::update_validator_manager(
+                        &chain_client,
+                        &validators,
+                        validator_manager_abi.clone(),
+                    )
+                    .await
+                    .is_err()
+                    {
+                        error!(
```

Comment:
Please print the error in this case - `if let Err(e) = Self::update_validator_manager(...).await { error!(?e, ...); }`